### ISSUE TITLE
batch out of range & loss value becomes 'nan' when running monocular depth estimation

### ISSUE THREAD

**Issue by Gacha76** [keras-team/keras-io#1832]
### Issue Type

Bug

### Source

source

### Keras Version

Keras 2.10

### Custom Code

No

### OS Platform and Distribution

Windows 11

### Python version

3.10.13

### GPU model and memory

RTX 3050 6GB

### Current Behavior?

When calling the `.fit()` function to train the model, the 1st epoch runs as expected and stops when all batches have been iterated.

The problem starts from the 2nd epoch onwards where batches start running out of the given range and loss values become `nan`. Once the epoch is complete, the UI becomes normal again but this behavior is observed again for the 3rd epoch and so on.

![Screenshot 2024-04-13 111550](https://github.com/keras-team/keras-io/assets/114499152/0bc8967e-9160-48da-8577-83f3404fe9ca)

All tutorials on Youtube running the same Colab notebook given by the [Keras Team](https://keras.io/examples/vision/depth_estimation/) seem to run without having any issues and the model trains properly but this isn't the case when I run the notebook locally or on Colab using both CPU and GPU.

### Standalone code to reproduce the issue or tutorial link

```shell
https://colab.research.google.com/github/keras-team/keras-io/blob/master/examples/vision/ipynb/depth_estimation.ipynb
```

**Comment by sachinprasadhs** [comment]
You seem to be using the older `Keras` version in your local system. `Keras 3` with multi backend feature is available, you can upgrade the Keras package and try again.

`pip install -U keras`

**Comment by Gacha76** [comment]
Below screenshot is using `Keras 3`. Batch no longer goes out of range but loss values still become `nan`.

![Screenshot 2024-04-16 164217](https://github.com/keras-team/keras-io/assets/114499152/0dbdd7ff-0582-400a-9f35-44596fa94d50)

**Comment by sachinprasadhs** [comment]
The tutorial which you are referring to has not been migrated to Keras 3 yet, possibly due to some dependency on Tensorflow or Keras 2 APIs.

I was able ti run the tutorial successfully for 1 epoch with TensorFlow 2.15 which uses Keras 2.15 in it's backend.
Attaching the working Gist here for reference https://colab.sandbox.google.com/gist/sachinprasadhs/5aead85438db273c01e72ec257d6c09e/depth_estimation.ipynb

**Comment by Gacha76** [comment]
It works for 1 epoch for me as well in both Keras 2 and Keras 3. The issue arises when I need to train the model for more than 1 epoch which results in the above behavior. Also, since the loss values become undefined, the neural network starts to output nothing but a black screen as shown here.

![Screenshot 2024-04-17 090156](https://github.com/keras-team/keras-io/assets/114499152/37a9d47a-ce93-4f21-8dfd-b6641ae3119b)

**Comment by sachinprasadhs** [comment]
In the published tutorial we can see output for more number of epochs.
Since the tutorial is not yet migrated to Keras 3, we can look at it once the tutorial is migrated to Keras 3, Keras team doesn't have enough bandwidth to migrate tutorials.
Community contributions are welcome.

**Comment by sonali-kumari1** [comment]
Hi @Gacha76 -

Thanks for reporting this issue. The [Monocular depth estimation example](https://github.com/keras-team/keras-io/blob/master/examples/vision/depth_estimation.py) was migrated to Keras 3 (with Tensorflow backend) in commit https://github.com/keras-team/keras-io/commit/ff969feea1dcbe081c3bb9fb468fb46825246936. 

I have tested the example with the latest version of Keras(3.11.3) and Tensorflow(2.20.0) and it works fine in this [gist](https://colab.sandbox.google.com/gist/sonali-kumari1/a3fdd644839884b12c86da08c5b4384f/-1832-depth_estimation.ipynb). The model no longer produces nan values for loss and the predicted depth map images seems fine. 

**Loss over epochs:**

<img width="1276" height="716" alt="Image" src="https://github.com/user-attachments/assets/d54ed519-2497-4576-acc6-52d25853605b" />

**Predicted depth map:**

<img width="2652" height="1000" alt="Image" src="https://github.com/user-attachments/assets/305e35fc-2f64-49e4-8d43-b44ee5d5dc5b" />
